### ISSUE TITLE
S3: Starting upload of partition objects, thread 'tokio-runtime-worker' panicked at 'not stream'

### ISSUE THREAD

**Issue by jychen7** [splitgraph/seafowl#253]
[Not urgent] hi team, I am just trying the tutorial with S3 object store (my private bucket) and hit this error. I would like to provide more information if need.

### Reproduce Step
seafowl version: 0.2.9 (I also try 0.2.8 and also fail with same error)
os: macOS
binary: [seafowl-v0.2.9-x86_64-apple-darwin.tar.gz](https://github.com/splitgraph/seafowl/releases/download/v0.2.9/seafowl-v0.2.9-x86_64-apple-darwin.tar.gz)

```
curl -i \
    -H "Content-Type: application/json" \
    -H "Authorization: Bearer ******" \
localhost:8080/q -d@-<<EOF
{"query": "INSERT INTO test_cache
  VALUES
    ('Alpha', 2.3, 17),
    ('Alpha', 2.4, 18),
    ('Beta', 9.3, 42),
    ('Beta', 4.0, 18),
    ('Gamma', 6.8, 2),
    ('Gamma', 1.2, 8)
"}
EOF
```

### Expect

HTTP/1.1 200 OK

### Actual

HTTP/1.1 400 Bad Request

```
seafowl::context  > Starting upload of partition objects
thread 'tokio-runtime-worker' panicked at 'not stream', /Users/runner/.cargo/registry/src/github.com-1ecc6299db9ec823/object_store-0.5.1/src/aws/credential.rs:172:27
```

### seafowl config

```
[object_store]
type = "s3"
access_key_id = "******"
secret_access_key = "******"
endpoint = "s3.us-east-1.amazonaws.com"
bucket = "******"

# Store the catalog on the local disk
[catalog]
type = "sqlite"
dsn = "seafowl-data/seafowl.sqlite"

# Configure the HTTP frontend
[frontend.http]
bind_host = "127.0.0.1"
bind_port = 8080

# By default, make Seafowl readable by anyone...
read_access = "any"
write_access = "******"
```

**Comment by mildbyte** [comment]
@jychen7 Thanks for the report!

Can you check if this still fails on the latest nightly build (https://nightly.link/splitgraph/seafowl/workflows/nightly/main) where we upgraded the `object_store` crate to 0.5.3 to see if this fixes it?

If it's still breaking, can you rerun with the `RUST_LOG=debug` environment variable set and paste the log when the failure happens?

**Comment by jychen7** [comment]
> where we upgraded the object_store crate to 0.5.3 to see if this fixes it?

looks like latest version (as of 2022-12-29) of object_store is 0.5.2. And current main 455da47784d55c7e742580b1616cc1d1effa46d1 is refer to 0.5.2 as well, not 0.5.3?
https://github.com/splitgraph/seafowl/blob/455da47784d55c7e742580b1616cc1d1effa46d1/Cargo.lock#L3684-L3686

> If it's still breaking, can you rerun with the RUST_LOG=debug environment variable set and paste the log when the failure happens?

@mildbyte yes, I try the nightly build and `seaflow -v` shows this commit 455da47784d55c7e742580b1616cc1d1effa46d1

`RUST_LOG=debug` shows
```
 2022-12-29T23:51:49.829Z DEBUG datafusion::execution::disk_manager   > Created directory '"/var/folders/qx/2dsbtw0d7cn9fm4l4pghl9_00000gn/T/.tmpTx2rHU"' as DataFusion tempfile directory for Open a temporary file to write partition
 2022-12-29T23:51:49.829Z DEBUG datafusion::physical_plan::projection > Start ProjectionExec::execute for partition 0 of context session_id 7119bccd-0b45-4f50-9a79-386409a2f5d2 and task_id None
 2022-12-29T23:51:49.833Z INFO  seafowl::context                      > Starting upload of partition objects
thread 'tokio-runtime-worker' panicked at 'not stream', /Users/runner/.cargo/registry/src/github.com-1ecc6299db9ec823/object_store-0.5.2/src/aws/credential.rs:173:27
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
 2022-12-29T23:51:49.834Z INFO  seafowl::frontend::http               > 127.0.0.1:64118 "POST /q HTTP/1.1" 400 "-" "curl/7.77.0" 13.33997ms
 2022-12-29T23:51:49.834Z DEBUG hyper::proto::h1::io                  > flushed 162 bytes
 2022-12-29T23:51:49.834Z DEBUG hyper::proto::h1::conn                > read eof
```

`RUST_LOG=debug RUST_BACKTRACE=1` suggest to use `RUST_LOG=debug RUST_BACKTRACE=full`
`RUST_LOG=debug RUST_BACKTRACE=full` shows
```
2022-12-30T00:00:39.890Z DEBUG datafusion::execution::disk_manager   > Created directory '"/var/folders/qx/2dsbtw0d7cn9fm4l4pghl9_00000gn/T/.tmpihjeva"' as DataFusion tempfile directory for Open a temporary file to write partition
 2022-12-30T00:00:39.890Z DEBUG datafusion::physical_plan::projection > Start ProjectionExec::execute for partition 0 of context session_id bed2421e-2af2-4048-bf85-5fd82b4bd623 and task_id None
 2022-12-30T00:00:39.894Z INFO  seafowl::context                      > Starting upload of partition objects
thread 'tokio-runtime-worker' panicked at 'not stream', /Users/runner/.cargo/registry/src/github.com-1ecc6299db9ec823/object_store-0.5.2/src/aws/credential.rs:173:27
stack backtrace:
   0:        0x105074387 - __mh_execute_header
   1:        0x1037e42fa - __mh_execute_header
   2:        0x105046dbc - __mh_execute_header
   3:        0x105077d9a - __mh_execute_header
   4:        0x1050779c6 - __mh_execute_header
   5:        0x105078a23 - __mh_execute_header
   6:        0x105078504 - __mh_execute_header
   7:        0x105078468 - __mh_execute_header
   8:        0x105078432 - __mh_execute_header
   9:        0x105567293 - ___jit_debug_register_code
  10:        0x1037e8bdb - __mh_execute_header
  11:        0x1037e8b8c - __mh_execute_header
  12:        0x105567409 - ___jit_debug_register_code
  13:        0x1045a4aa0 - __mh_execute_header
  14:        0x1045e415b - __mh_execute_header
  15:        0x104e38a2b - __mh_execute_header
  16:        0x104e33f71 - __mh_execute_header
  17:        0x104e3356e - __mh_execute_header
  18:        0x1050a80d7 - __mh_execute_header
  19:        0x1050a5f30 - __mh_execute_header
  20:        0x10509193f - __mh_execute_header
  21:        0x105091682 - __mh_execute_header
  22:        0x10507bc99 - __mh_execute_header
  23:     0x7ff812f314f4 - __pthread_start
 2022-12-30T00:00:39.895Z INFO  seafowl::frontend::http               > 127.0.0.1:64169 "POST /q HTTP/1.1" 400 "-" "curl/7.77.0" 13.387037ms
 2022-12-30T00:00:39.895Z DEBUG hyper::proto::h1::io                  > flushed 162 bytes
 2022-12-30T00:00:39.896Z DEBUG hyper::proto::h1::conn                > read eof
```

**Comment by gruuya** [comment]
Thanks for the bug report and detailed logs @jychen7! After taking a closer look I realized the issue was that we weren't requiring the `region` param in the `object_store` section, and in turn the object store didn't parse it from the `endpoint`. 

This is now fixed (with #255) in the latest Seafowl version (`0.2.10`). You can now pass only `region` when pointing to AWS proper, otherwise pass `endpoint` if you wish to use a different S3-compatible store (e.g. minio, localstack).

**Comment by jychen7** [comment]
Awesome. I confirm `0.2.10` works, thank you

```
[object_store]
type = "s3"
access_key_id = "******"
secret_access_key = "******"
region = "us-east-1"
bucket = "******"
```
